colossalai/inference/core/engine.py

```diff
@@ -445,15 +450,19 @@ def generate(
             List[str]: Inference result returned by one generation.
         """
         with torch.inference_mode():
+            if generation_config is not None:
+                self.generation_config = generation_config
+
             if prompts is not None or prompts_token_ids is not None:
+                if isinstance(prompts, str) and isinstance(request_ids, int):
+                    prompts = [prompts]
+                    request_ids = [request_ids]
                 self.add_request(request_ids=request_ids, prompts=prompts, prompts_token_ids=prompts_token_ids)
 
             output_seqs_list = []
             total_tokens_list = []
 
             # intuition: If user provide a generation config, we should replace the existing one.
-            if generation_config is not None:
-                self.generation_config = generation_config
```

Comment:
Why moving it to the beginning of the function. Please keep the comment with the operation. I'm assuming this is caused by your merging.

Reply:
fixed, caused by merging